### PR TITLE
HELP-37837: Fix tasks app startup race condition issue

### PR DESCRIPTION
**Pull request by harenson**

### CODE REVIEW

File: applications/tasks/src/knm_number_crawler.erl [hunk 1]
```diff
@@ -98,6 +98,9 @@ handle_call(_Request, _From, State) ->
 handle_cast('stop', State) ->
     lager:debug("crawler has been stopped"),
     {'stop', 'normal', State};
+handle_cast({'$proxy_stop', Reason}, State) ->
+    lager:debug("~p stopping with reason: ~p", [?MODULE, Reason]),
+    {stop, Reason, State};
```
Comment (jamesaimonetti) [inline applications/tasks/src/knm_number_crawler.erl]: plz tickie all the atoms

Comment (lazedo) [reply applications/tasks/src/knm_number_crawler.erl]: `handle_cast({'$proxy_stop', Reason}, State) ->` are these really necessary ?

File: applications/tasks/src/knm_port_request_crawler.erl [hunk 2]
```diff
@@ -82,6 +82,9 @@ handle_call(_Request, _From, State) ->
 handle_cast('stop', State) ->
     lager:debug("crawler has been stopped"),
     {'stop', 'normal', State};
+handle_cast({'$proxy_stop', Reason}, State) ->
```
Comment (lazedo) [inline applications/tasks/src/knm_port_request_crawler.erl]: same here, will skip the others

File: core/kazoo_globals/src/kz_globals.erl [hunk 3]
```diff
@@ -574,6 +581,10 @@ do_amqp_unregister(Global, Reason) ->
               ],
     lager:debug("deleting ~p", [Name]),
     ets:delete(?TAB_NAME, Name),
+    kz_util:spawn(fun() ->
+                          timer:sleep(500), % Give time to other nodes to register the name
+                          gen_listener:call(?SERVER, {'amqp_delete', Global, Reason})
+                  end),
```
Comment (lazedo) [inline core/kazoo_globals/src/kz_globals.erl]: is this necessary ?

File: core/kazoo_globals/src/kz_globals.erl [hunk 4]
```diff
@@ -150,8 +150,13 @@ where_is(Name) ->
 -spec lookup_name(kz_global:name()) -> kz_global:global() | 'undefined'.
 lookup_name(Name) ->
     case ets:lookup(?TAB_NAME, Name) of
-        [Global] -> Global;
-        [] -> 'undefined'
+        [Global] ->
+            case is_process_alive(kz_global:pid(Global)) of
+                'true' -> Global;
+                'false' -> 'undefined'
```
Comment (icehess) [inline core/kazoo_globals/src/kz_globals.erl]: If process is not alive shouldn't we delete it from ETS table?

**Comment by jamesaimonetti** [comment]
@harenson can you rebase and address @lazedo's comments?

**Comment by lazedo** [comment]
@harenson i think the only change that makes sense is the `lookup_name`. the other changes need explanation of their need.

**Pull request merged**